app/models/rbac.rb

```diff
@@ -496,7 +496,7 @@ def self.search(options = {})
       auth_count  = targets.length
     end
 
-    unless search_filter.nil?
+    if search_filter && targets && (!exp_attrs || !exp_attrs[:supported_by_sql])
```

Comment:
@kbrock I was about to comment on the your previous version of this logic and suggest using this. I just want to double check that supported_by_sql is properly set before I can be comfortable with this.